### PR TITLE
[Merged by Bors] - chore: Golf `prod_involution`

### PR DESCRIPTION
**Pull request by YaelDillies**
... and extract `prod_ite_zero` from the proof of `prod_boole`

Co-authored-by: Bhavik Mehta, <bhavik.mehta8@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/BigOperators/Group/Finset.lean [hunk 1]
```diff
@@ -1511,44 +1511,34 @@ theorem prod_flip {n : ℕ} (f : ℕ → β) :
...
+    exact ih (s \ {x, g x hx}) (ssubset_iff.2 ⟨x, by simp [insert_subset_iff, hx]⟩) _
+      (by simp [hg₁]) (fun _ _ => hg₃ _ _) h₃ (fun _ _ => hg₄ _ _)
+  simp only [mem_sdiff, mem_insert, mem_singleton, not_or, g_mem, true_and]
+  rintro a ⟨ha₁, ha₂, ha₃⟩
+  refine ⟨fun h => by simp [← h, hg₄] at ha₃, fun h => ?_⟩
+  have : g (g a ha₁) (g_mem _ _) = g (g x hx) (g_mem _ _) := by simp only [h]
+  exact ha₂ (by simpa [hg₄] using this)
+
+@[to_additive]
+lemma prod_ninvolution (g : α → α) (hg₁ : ∀ a, f a * f (g a) = 1) (hg₂ : ∀ a, f a ≠ 1 → g a ≠ a)
```
Comment (eric-wieser) [inline Mathlib/Algebra/BigOperators/Group/Finset.lean:1538]: Can you add a docstring explaining the `n`?

Comment (eric-wieser) [reply Mathlib/Algebra/BigOperators/Group/Finset.lean:1538]: (and for the additive one)

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**